tests/model_selection/test_hyperband.py

```diff
@@ -0,0 +1,247 @@
+import numpy as np
+import scipy.stats
+from sklearn.linear_model import SGDClassifier
+
+import dask
+import dask.array as da
+from dask.distributed import Client, wait
+from distributed.utils_test import loop, cluster, gen_cluster  # noqa: F401
+from tornado import gen
+
+from dask_ml.datasets import make_classification
+from dask_ml.model_selection import HyperbandCV
+from dask_ml.wrappers import Incremental
+from dask_ml.model_selection._hyperband import _partial_fit
+from dask_ml.utils import ConstantFunction
+from distributed.metrics import time
+
+import pytest
+
+
+@pytest.mark.parametrize("array_type,library",  # noqa: F811
+                         [("dask.array", "dask-ml"),
+                          ("numpy", "sklearn"),
+                          ("numpy", "test")])
+def test_sklearn(array_type, library, loop, max_iter=27):
+    with cluster() as (s, [a, b]):
+        with Client(s['address'], loop=loop):
+            chunk_size = 100  # make dask array with one chunk
+            d = 20
+            X, y = make_classification(n_samples=chunk_size, n_features=d,
+                                       random_state=42, chunks=chunk_size)
+            if array_type == "numpy":
+                X = X.compute()
+                y = y.compute()
+                chunk_size = X.shape[0]
+
+            sgd_params = {'alpha': np.logspace(-2, 1, num=1000),
+                          'l1_ratio': np.linspace(0, 1, num=1000),
+                          'average': [True, False]}
+            kwargs = dict(tol=-np.inf, penalty='elasticnet', random_state=42)
+            if library == "sklearn":
+                model = SGDClassifier(**kwargs)
+                params = sgd_params
+            elif library == "dask-ml":
+                model = Incremental(SGDClassifier(**kwargs))
+                params = sgd_params
+            elif library == "test":
+                model = ConstantFunction()
+                params = {'value': np.linspace(0, 1, num=1000)}
+            else:
+                raise ValueError
+
+            search = HyperbandCV(model, params, max_iter=max_iter,
+                                 random_state=42, asynchronous=False)
+            search.fit(X, y, classes=da.unique(y))
+
+            score = search.best_estimator_.score(X, y)
+            if library == "sklearn":
+                assert score > 0.4
+            if library == "dask-ml":
+                assert score > 0.2
```

Comment:
These scores seem quite low.  Is there a reason why?  Do we have confidence that Hyperband works well on real problems?  Has anyone tried it on something non-trivial?

Reply:
I've increased the scores it's testing for. I had these so low to account for worst case scenarios because the tests are deterministic.